### PR TITLE
Add support for images

### PR DESCRIPTION
**Pull request by eperedo**
### :pushpin: References

* **Issue:** Closes https://app.clickup.com/t/865cm2c33

### :memo: Implementation

I tried to use d2-api, [but the domain is hardcoded to **"DOCUMENT"**](https://github.com/EyeSeeTea/d2-api/blob/master/src/api/files.ts#L43) and for this case it must be **"DATA_VALUE"**

### :fire: Notes for the reviewer

Zip file for testing: [test-full.zip](https://github.com/EyeSeeTea/Bulk-Load/files/11886882/test-full.zip)

Docker: [docker.eyeseetea.com/eyeseetea/dhis2-data:2.36.11.1-sp-ip-test](docker.eyeseetea.com/eyeseetea/dhis2-data:2.36.11.1-sp-ip-test)

### :art: Screenshots

**Excel with filenames for before/after datavalue**
excel-file-names

**Support for zip files**
zip-support

**Tracker with images from zip file**
tracker-event-images

### CODE REVIEW

**Review by tokland — COMMENTED** [review]
The feature worked for me (in a first import). Check the comments for some doubts/proposals.

File: package.json
[the remark this answers is not in the document]
Comment (marballes) [reply package.json:1]: @tokland It was the initial idea but then it become a new functionality

File: src/data/FileD2Repository.ts [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+};
+
+export class FileD2Repository implements FileRepository {
+    private api: D2Api;
+
+    constructor(localInstance: DhisInstance) {
+        this.api = new D2ApiDefault({ baseUrl: localInstance.url });
+    }
+
+    async uploadAll(files: FileResource[]): Promise<FileResource[]> {
```
[the remark this answers is not in the document]
Comment (eperedo) [reply src/data/FileD2Repository.ts:24]: There's not an easy way to reuse images, in any case if files are not related to a datavalue DHIS will delete them automatically.

https://docs.dhis2.org/en/full/develop/dhis-core-version-236/developer-manual.html#webapi_file_resources_constraints:~:text=The%20ID%20returned,cleaned%20up%20automatically.

**Review by tokland — APPROVED** [review]
All good!

**Pull request merged**